docs/spec/nmt-wrapper.md

```diff
@@ -0,0 +1,83 @@
+The rows and columns of the original data square are then extended using a 2D Reed-Solomon coding scheme.
+The [extended version](https://github.com/celestiaorg/celestia-app/blob/specs-staging/specs/src/specs/data_structures.md#2d-reed-solomon-encoding-scheme) is called extended data square, that is a `2k` by `2k` square consisting of `4` quadrants namely, `Q0`, `Q1`, `Q2`, and `Q3`.
+Figure 1 provides an illustration of a sample data square and its extended version.
+`Q0` corresponds to the original data square.
+`Q1` and `Q2` represent the horizontal and vertical extensions of `Q0`, respectively. 
+`Q3` is the extension of `Q2` in the horizontal direction.
+```markdown
+  k         k
+ -------   -------
+|        |        |
+|   Q0 → |   Q1   | k
+|   ↓    |        |
+-------  -------
+|        |        |
+|   Q2 → |   Q3   | k
+|        |        |
+ -------   -------
+```
+Figure 1.
+
+Each row and column of the extended data square is modeled by a [Namespace Merkle Tree](https://github.com/celestiaorg/nmt/blob/master/spec/nmt.md).
+NMTs require the data items they represent to be namespaced, which means that the shares within each row or column of the extended data square must be namespaced before being added to the NMT.
+This is where the [Namespaced MerkleTree Wrapper](https://github.com/celestiaorg/celestia-app/blob/main/pkg/wrapper/nmt_wrapper.go)  comes into play.
+It is a data structure that wraps around the [Namespaced Merkle Tree](https://github.com/celestiaorg/nmt/blob/master/spec/nmt.md) and ensures the proper namespaces are prepended to the shares  before they are added to their respective row or column NMT.
+In this specification, we elaborate on the design and structure of the Namespace Merkle Tree wrapper.
+
+
+
+## Namespaced Merkle Tree Wrapper
+Namespaced Merkle Tree Wrapper is used to represent a row or column of Celestia extended data square.
+At its core, it is an [NMT](https://github.com/celestiaorg/nmt/blob/master/spec/nmt.md) and is defined by a similar set of parameters.
+Namely, the [namespace ID size `NamespaceIDSize`](https://github.com/celestiaorg/nmt/blob/master/spec/nmt.md#nmt-data-structure), 
+the underlying hash function for the digest calculation of the [namespaced hashes](https://github.com/celestiaorg/nmt/blob/master/spec/nmt.md#namespaced-hash), 
+and the [`IgnoreMaxNamespace` flag](https://github.com/celestiaorg/nmt/blob/master/spec/nmt.md#ignore-max-namespace) which dictates how namespace range of non-leaf nodes are calculated.
+In addition, the NMT wrapper is configured with the data square size `SqaureSize` (`k` in the above example), and the index of the row or column it represents `AxisIndex` which is a value in `[0, SquareSize)`.
```

Comment:
[silly question] I see that `axisIndex` is the row or column index but I don't see something in `ErasuredNamespacedMerkleTree` that indicates if it is a row or column. I don't understand how the erasured NMT knows which quadrant a leaf is being pushed to. 

For example if `squareSize=4`, `axisIndex=3`, and the first first leaf is being pushed to an NMT. How does the NMT know if it is pushing to Q1 (if this is column 3 of EDS) or Q2 (if this is row 3 of EDS)? Perhaps it doesn't matter because in either case this is parity data but I'm curious if it matters. If the NMT doesn't know which quadrant is being pushed, then I think this [godoc](https://github.com/celestiaorg/celestia-app/blob/514c8dd37ac2c4dcee0c653d20f1d9020f34075a/pkg/wrapper/nmt_wrapper.go#L28-L29) is misleading.

[nit] since the fields are unexported, proposal to use the lowercase versions so that the variable here matches the variable in the file. Also typo in squareSize:

```suggestion
In addition, the NMT wrapper is configured with the data square size `squareSize` (`k` in the above example), and the index of the row or column it represents `axisIndex` which is a value in `[0, SquareSize)`.
```

Reply:
I think my silly question can be disregarded b/c later in this document it states

> During the insertion, the wrapper first checks whether the share is within the original data square or not.

so the wrapper doesn't actually know which quadrant the leaf is being pushed to.

Reply:
Good question and correct answer! The NMT wrapper is designed to only identify if shares fall within Q0. If it were to identify shares in other quadrants, it would require a row or column indicator to distinguish shares in Q1 from those in Q3.

Reply:
> since the fields are unexported, proposal to use the lowercase versions so that the variable here matches the variable in the file.

Thank you for the suggestion. My intention was to describe the wrapper as a general data structure without being specific to its implementation, which is why the capitalization did not match the implementation (this was intentional).